### DISCUSSION TITLE
How do I set global SASS rules from a plugin?

### DISCUSSION THREAD

**Discussion by stapelberg** [pretix/pretix#3249] [Q&A] [answered]
Currently, I’m using the following patch:

```patch
--- a/src/pretix/presale/style.py
+++ b/src/pretix/presale/style.py
@@ -92,6 +92,10 @@ def compile_scss(object, file="main.scss", fonts=True):
                 font
             ))
 
+    if font == 'Alegreya Sans' and fonts:
+        sassrules.append('$font-size-base:          18px !default;')
+        sassrules.append('$headings-font-weight:    bold !default;')
+
     if isinstance(object, Event):
         for recv, resp in sass_preamble.send(object, filename=file):
             sassrules.append(resp)
```

…but I would like to get rid of it in favor of injecting these SASS rules from the plugin which implements the rest of the design (static CSS, register_fonts and global_html_page_header).

I found the https://docs.pretix.eu/en/latest/development/api/general.html#pretix.presale.signals.process_request signal, but it seems to only be sent in the scope of Events, i.e. does not style the login form, for example: https://github.com/pretix/pretix/blob/beea439df890946afd17b4560ee11b4cd5004e3c/src/pretix/presale/style.py#L95-L97

Is there a signal I didn’t see? Should a new one be added? Or should the existing style.py code be modified?

Thanks

**Comment by raphaelm** [comment, marked as answer]
**First of all,** we intentionally not make re-styling of pretix easy. It's something we do not want to encourage or support. We offer some options for design customizations (colors, fonts, round edges), and everything else is out of scope.

`sass_preamble` and `sass_postamble` *can* be used for this, yes, and we have ourselves abused them for this in one or two cases before, but we try very hard not too. They came into existance for a different purpose, I think we needed them to develop frontend-heavy plugins like the seating module.

There is no similar signal on organizer level, since there is no concept of plugins on organizer level, and there is no plugin activation/deactivation that would trigger the signal. (As a side note, we would very much like to have plugins at organizer level at some point in which case it would make sense to add signals for that, but we have not even discussed a concept of what plugins on organizer level could look like and how they would interact with event level plugins.)

As any signal on organizer level would trigger for all organizers on the instance, you could just override the entire `variables.scss` file using the `static` folder of a plugin, but that wouldn't solve your "if font…" problem.

**That said,** using a different base font size for a specific font is something we **could** consider adding to API of the `register_fonts` signal. Is this just a matter of taste or is Alegreya Sans really visually considerably smaller than other fonts?

**Reply by stapelberg** [reply]
Thanks for your quick reply!

> **First of all,** we intentionally not make re-styling of pretix easy. It's something we do not want to encourage or support. We offer some options for design customizations (colors, fonts, round edges), and everything else is out of scope.

Understood. We might eventually let customers order using a custom frontend to get maximum control over the presentation. But for now, a limited custom theme (font, logo, css) is good enough :)

> There is no similar signal on organizer level, since there is no concept of plugins on organizer level, and there is no plugin activation/deactivation that would trigger the signal. (As a side note, we would very much like to have plugins at organizer level at some point in which case it would make sense to add signals for that, but we have not even discussed a concept of what plugins on organizer level could look like and how they would interact with event level plugins.)

Good to know! I was wondering about organizer-level plugins.

> **That said,** using a different base font size for a specific font is something we **could** consider adding to API of the `register_fonts` signal. Is this just a matter of taste or is Alegreya Sans really visually considerably smaller than other fonts?

Alegreya Sans is not considerably smaller, this is just a matter of preference (or, rather, consistency with our existing style on other web properties).

Adding the font size to the register_fonts signal sounds good to me. Should I file a feature request for this? I might also be able to supply a PR, though I’m not sure when exactly.

Regarding the other half of my patch: you might want to make `$headings-font-weight: bold` the default for user-supplied fonts — I think the current default `$headings-font-weight: 500` doesn’t generally work well when not using Open Sans.
